src/lib.rs

```diff
@@ -155,16 +155,24 @@ impl CodecPipelineImpl {
     }
 
     fn py_untyped_array_to_array_object<'a>(
-        value: &Bound<'a, PyUntypedArray>,
+        value: &'a Bound<'_, PyUntypedArray>,
     ) -> &'a PyArrayObject {
         let array_object_ptr: *mut PyArrayObject = value.as_array_ptr();
-        unsafe {
-            // SAFETY: array_object_ptr cannot be null
-            &*array_object_ptr
-        }
+        let array_object: &'a PyArrayObject = unsafe {
+            // SAFETY: array_object_ptr cannot be null and the array object pointed to by array_object_ptr is valid for 'a
+            array_object_ptr
+                .as_ref()
+                .expect("pointer is convertible to a reference")
+        };
+        array_object
     }
 
-    fn nparray_to_slice<'a>(value: &'a Bound<'_, PyUntypedArray>) -> &'a [u8] {
+    fn nparray_to_slice<'a>(value: &'a Bound<'_, PyUntypedArray>) -> Result<&'a [u8], PyErr> {
+        if !value.is_c_contiguous() {
+            return Err(PyErr::new::<PyValueError, _>(
+                "input array must be a C contiguous array".to_string(),
+            ));
+        }
```

Comment:
why is this the case?

Reply:
The elements represented by [`zarrs::array::ArrayBytes`](https://docs.rs/zarrs/latest/zarrs/array/enum.ArrayBytes.html) must be C-contiguous, I should probably document that.

Reply:
Yeah, but getting the data as a slice isn’t affected by that, so the check shouldn’t be inside of the function that does that, right?

Reply:
I could rename it to something like `nparray_with_elements_in_c_contiguous_order_to_bytes_slice` :). But let's just leave it for now. I suspect this area of the codebase may need some changes later when supporting variable length data types anyway.